assets/js/modules/analytics/datastore/accounts.js

```diff
@@ -80,12 +85,104 @@ export const actions = {
 			type: RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_FAILED,
 		};
 	},
+
+	/**
+	 * Creates a new Analytics account.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} args              Argument params.
+	 * @param {string} args.accountName  Google Analytics account name.
+	 * @param {string} args.propertyName Google Analytics property name.
+	 * @param {string} args.profileName  Google Analytics profile name.
+	 * @param {string} args.timezone     Google Analytics timezone.
+	 * @return {Function} Generator function action.
+	 */
+	*createAccount( { accountName, propertyName, profileName, timezone } ) {
+		invariant( accountName, 'accountName is required.' );
+		invariant( propertyName, 'propertyName is required.' );
+		invariant( profileName, 'profileName is required.' );
+		invariant( timezone, 'timezone is required.' );
+
+		const accountTicket = yield actions.fetchCreateAccount( { accountName, propertyName, profileName, timezone } );
+
+		return actions.receiveCreateAccount( { accountTicket } );
+	},
+
+	*fetchCreateAccount( { accountName, propertyName, profileName, timezone } ) {
+		let response;
+
+		yield {
+			payload: {},
+			type: START_FETCH_CREATE_ACCOUNT,
+		};
+
+		try {
+			response = yield {
+				payload: { accountName, propertyName, profileName, timezone },
+				type: FETCH_CREATE_ACCOUNT,
+			};
+
+			yield {
+				payload: {},
+				type: FINISH_FETCH_CREATE_ACCOUNT,
+			};
+		} catch ( error ) {
+			return actions.receiveCreateAccountFailed( { accountName, error } );
+		}
+		return response;
+	},
+
+	/**
+	 * Adds an account ticket to process.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} args               Argument params.
+	 * @param {Object} args.accountTicket Google Analytics create account ticket object.
+	 */
+	receiveCreateAccount( { accountTicket } ) {
+		invariant( accountTicket, 'accountTicket is required.' );
+
+		// Once we have an account ticket, redirect the user to accept the Terms of Service.
+		const { id } = accountTicket;
+		if ( id ) {
+			// Use `location.assign` so we can test this action in Jest.
+			location.assign( `https://analytics.google.com/analytics/web/?provisioningSignup=false#management/TermsOfService/&api.accountTicketId=${ id }` );
+		}
+	},
+
+	/**
+	 * Logs an error with account creation.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} args       Argument params.
+	 * @param {Object} args.error Error object.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveCreateAccountFailed( { error } ) {
```

Comment:
This function name is now "out of sync" with the action type it dispatches `CATCH_FETCH_CREATE_ACCOUNT`. We don't have this type of function elsewhere in our code per latest conventions, so it'd be easiest to remove this function entirely and move the return statement for `CATCH_FETCH_CREATE_ACCOUNT` directly into the `catch` clause within `fetchCreateAccount()`.

Reply:
removed & moved return in https://github.com/google/site-kit-wp/pull/1390/commits/33abf75cfa23406300c0c0c0f39504f7ba26e128